examples/get_event_messages.c

```diff
@@ -46,12 +48,31 @@ void callback(event_client_event_t *event) {
 void process_event_data(event_client_event_t *event) {
   if (!strcmp(event->topic, "milestones/latest")) {
     milestone_latest_t res = {};
-    parse_milestone_latest((char *)event->data, &res);
-    printf("Index :%u\nTimestamp : %lu\n", res.index, res.timestamp);
+    if (parse_milestone_latest((char *)event->data, &res) == 0) {
+      printf("Index :%u\nTimestamp : %lu\n", res.index, res.timestamp);
+    }
   } else if (!strcmp(event->topic, "milestones/confirmed")) {
     milestone_confirmed_t res = {};
-    parse_milestones_confirmed((char *)event->data, &res);
-    printf("Index :%u\nTimestamp : %lu\n", res.index, res.timestamp);
+    if (parse_milestones_confirmed((char *)event->data, &res) == 0) {
+      printf("Index :%u\nTimestamp : %lu\n", res.index, res.timestamp);
+    }
+  } else if (!strcmp(event->topic, "messages/referenced")) {
+    msg_metadata_t *res = res_msg_metadata_new();
+    if (res) {
+      if (parse_messages_metadata((char *)event->data, res) == 0) {
+        printf("Msg Id :%s\n", res->msg_id);
+        size_t parents_count = res_msg_metadata_parents_len(res);
+        for (size_t i = 0; i < parents_count; i++) {
+          printf("Parent Id %zu : %s\n", i + 1, res_msg_metadata_parent_get(res, i));
+        }
+        printf("Inclusion State : %s\n", res->inclusion_state);
+        printf("Is Solid : %s\n", res->is_solid ? "true" : "false");
+        printf("Should Promote : %s\n", res->should_promote ? "true" : "false");
+        printf("Should Reattach : %s\n", res->should_reattach ? "true" : "false");
+        printf("Referenced Milestone : %ld\n", res->referenced_milestone);
+      }
+      res_msg_metadata_free(res);
+    }
```

Comment:
is there a proper way to handle errors?
we can simply clean up and terminate the program when errors occur.

Reply:
Done.